test/e2e/performanceprofile/functests/10_performance_ppc/ppc.go

```diff
@@ -0,0 +1,200 @@
+package __performance_ppc
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	testlog "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/log"
+	"sigs.k8s.io/yaml"
+)
+
+type PPCTest struct {
+	PodmanMakeOptions func(args []string) []string
+	PodmanBinary      string
+}
+
+type PPCSession struct {
+	*gexec.Session
+}
+
+func (p *PPCTest) MakeOptions(args []string) []string {
+	return p.PodmanMakeOptions(args)
+}
+
+func (p *PPCTest) PodmanAsUserBase(args []string, noEvents, noCache bool) (*PPCSession, error) {
+	var command *exec.Cmd
+	podmanOptions := p.MakeOptions(args)
+	podmanBinary := p.PodmanBinary
+	fmt.Printf("Running: %s %s\n", podmanBinary, strings.Join(podmanOptions, " "))
+	command = exec.Command(podmanBinary, podmanOptions...)
+	session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+	if err != nil {
+		return nil, err
+	}
+
+	return &PPCSession{session}, nil
+}
+
+type PPCTestIntegration struct {
+	PPCTest
+}
+
+type PPCSessionIntegration struct {
+	*PPCSession
+}
+
+func (p *PPCTestIntegration) MakeOptions(args []string) []string {
+	return args
+}
```

Comment:
if this returns the same args we passed, then why do we need a method? is it for readability (just guessing) only?

Reply:
Currently it may seem to be doing nothing, but i want to extend this in future, for example some of the default options i can put it here and in the actual test, i can pass the actual options that are required for test.  Also i can use this function exclusively for all the podman options.